source/mainnet/smart-contracts/onboarding-guide-ethereum-developers/faq.rst

```diff
@@ -0,0 +1,371 @@
+
+    No. The `init` function has to be called by an account (not a smart contract) on the Concordium chain.
+
+.. dropdown::  Can I create a factory smart contract on Concordium?
+
+    No. A factory smart contract on the Ethereum chain deploys other smart contracts. In contrast,
+    the `init` function has to be called by an account (not a smart contract) on the Concordium chain.
+
+.. dropdown::  Can I predict/calculate the address of the smart contract before deploying it? Is there something similar to the Ethereum CREATE2?
+
+    Contract addresses on Concordium are represented by an index and a subindex as seen below.
+    When invoking the `init` function, a new smart contract instance is
+    deployed and assigned the next index number in sequential order.
+    The subindex is currently not in use and is always 0. There are plans to give the subindex meaning in the future.
+
+    In contrast to Ethereum, you cannot send CCD to a contract address (or account address) before they have been deployed/initialized.
+
+    .. code-block:: console
+
+        ContractAddress {
+            index:    1234,
+            subindex: 0,
+        };
+
+.. dropdown::  How can I deploy a smart contract to the Concordium chain?
+
+    You can follow the chapter :ref:`deploying a smart contract<piggy-bank-deploying>` in the piggy bank tutorial.
+
+Concordium tools:
+=================
+
+.. dropdown::  Can I upload and verify my smart contract code on the block explorer (CCDScan)?
+
+    CCDScan currently does not support compiling, hosting, or verifying your smart contract code.
+    You are welcome to publish your smart contract code in public source code management tools such as `GitHub <https://github.com/>`_.
+
+Miscellaneous:
+==============
+
+.. dropdown:: What is the native currency on Concordium?
+
+    The native currency of the Concordium chain is CCD.
+
+.. dropdown:: Where do I get some test CCD? Is there a testnet faucet?
+
+    There are several options to request test CCD:
+
+    **Option 1:**
+    If you just created your account in the mobile app wallet then you
+    find a button to request 2000 testnet CCD to get started with your new account.
+
+    .. image:: ../tutorials/piggy-bank/images/pb_tutorial_5.png
+        :width: 20 %
+    .. image:: ../tutorials/piggy-bank/images/pb_tutorial_6.png
+        :width: 20 %
+
+    **Option 2:** If you have the curl package installed on your Unix-like operating systems,
+    you can request CCD in the terminal directly from the wallet proxy via the below command.
+
+    .. code-block:: console
+
+        $curl -X PUT https://wallet-proxy.testnet.concordium.com/v0/testnetGTUDrop/<YourAccountAddress>
+
+    If you insert your account address correctly, the command should look similar to the below line.
+
+    .. code-block:: console
+
+        $curl -X PUT https://wallet-proxy.testnet.concordium.com/v0/testnetGTUDrop/4phD1qaS3U1nLrzJcgYyiPq1k8aV1wAjTjYVPE3JXBDAz9WdEy
```

Comment:
Maybe also add an example "good" response and example bad response (probably the main thing will be that they will request the drop more than once)

Reply:
A more user-friendly tool would also be Postman.